.github/workflows/docker.yml

```
@@ -0,0 +1,105 @@
+name: Build and publish Docker images
+
+on:
+  push:
+    branches:
+      - "**"
+    tags:
+      - "v*.*.*"
+  pull_request:
+    branches:
+      - "main"
+  workflow_run:
+    workflows: [Rebase Upstream]
+    types:
+      - completed
```

The repo uses GitHub releases, so you should probably hook tho this event.
```suggestion
on:
  release:
    types: [published]
```

---

I've added it. I think it can be valuable to have docker images generated on PRs, to easily test them, but it's hard to know what is required/desired by upstream.

---

FWIW, a project I maintain has generic/reusable workflows that other workflows call. This allows for a workflow to build the image that can be used by both our [release (publishing)](https://github.com/docker-mailserver/docker-mailserver/blob/e6d519b6f82f1d3a1fada4b24edaed6f64f5d3f0/.github/workflows/default_on_push.yml#L33-L39) and [PR (testing)](https://github.com/docker-mailserver/docker-mailserver/blob/e6d519b6f82f1d3a1fada4b24edaed6f64f5d3f0/.github/workflows/test_merge_requests.yml#L18-L28) workflows.

If you take that approach, then the maintainers can integrate with whatever workflow process suits them as they see fit :)